### DISCUSSION TITLE
setData貌似不会触发SchemaReactions?

### DISCUSSION THREAD

**Discussion by busy-dog** [alibaba/formily#2982] [Q&A 2.x]
在选择器组件中，由于选项ID(uuid)和业务code(数字)用了两个不同的字段，后端要求表单提交ID，但是字段联动要依赖code判断，所以我把code放在field.data中存储，但是在SchemaReactions中使用$self.data来判断时发现field.setData不会触发SchemaReactions。是设计如此吗，还是bug?
毕竟我在effect字段中没有找到onFieldDataChange这个trigger :D

**Comment by janryWang** [comment]
你改的是data.xx，你在reactions里依赖的是field.data，当然不会触发变化，多看看reactive文档，理解清楚什么是响应式吧

**Reply by busy-dog** [reply]
我之前是这么写的:
`const data = cloneDeep(source.codes);`
`const inputValue = source.texts?.join(',');`
`field.setState({ data, inputValue });`
然后我今天尝试了这个写法
`field.setData(null);`
`const data = cloneDeep(source.codes);`
`const inputValue = source.texts?.join(',');`
`field.setState({ data, inputValue });`
还是没有触发reactive
我在哪里的理解出现了偏差呢，大佬可以推荐一些 [reactive] 相关的文档吗?

**Reply by VinVC** [reply]
你可以在codesandbox中展示一下你的用法。响应式可以参考[@formily/reactive](https://reactive.formilyjs.org/zh-CN/guide)和[mobx](https://cn.mobx.js.org/)
